experimental/telemetry/tracker.go

```diff
@@ -66,20 +99,42 @@ func NewTracker(
 		serverVersion:      serverVersion,
 		pluginID:           pluginID,
 		pluginVersion:      pluginVersion,
-		enabled:            enableDiagnostics,
+		enabledLogging:     config.EnabledLogging,
+		enabledTracking:    config.EnabledTracking,
+		logger:             l,
 	}
 }
 
+func (t *tracker) ReloadConfig(config TrackerConfig) {
+	if config.EnabledTracking != t.enabledTracking {
+		if config.EnabledTracking {
+			t.debugf("Enabling plugin telemetry")
+		} else {
+			t.debugf("Disabling plugin telemetry")
+		}
+	}
+
+	t.enabledTracking = config.EnabledTracking
+	t.enabledLogging = config.EnabledLogging
```

Comment:
These `ReloadConfig` calls might happen in quick succession. Do any precautionary measures need to be taken here?

Reply:
do you have previous experiences of onConfigurationChange called in a burst 
?
are you thinking of adding a lock? 

are you concerned about quick changes or about concurrent calls?

Reply:
There is nothing we can do about the quick changes. If the config changes, so does the tracker.

My question is about the concurrent calls. When race conditions happen, can they cause issues?

Reply:
I feel like overthinking a bit this but here we go:
- if concurrent calls are made with the same config, I don't see any potential issue (aside from some double login)
- if concurrent calls are made with different configs, we could end up with part of the config updated from one call and part from the other. To be honest, I don't know if that scenario can happen, but if that's the case I'd add a lock to make the changes atomically.


Do you have something in mind?

Reply:
I agree that a look should protect the writes to ensure no partial writes happen.